### PR TITLE
Check for root of receiver in selector expression

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/6373

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeVal.kt [hunk 1]
```diff
@@ -138,9 +144,9 @@ class VarCouldBeVal(config: Config) : Rule(
                 expression.left?.let(::visitAssignment)
 
                 // Check for whether the assignment contains an object literal.
-                val descriptor = (expression.left as? KtNameReferenceExpression)?.let {
-                    it.getResolvedCall(bindingContext)?.resultingDescriptor
-                }
+                val descriptor = (expression.left as? KtNameReferenceExpression)
+                    ?.getResolvedCall(bindingContext)
+                    ?.resultingDescriptor
```
Comment (cortinico) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeVal.kt:149]: Why are you changing this?

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/NullCheckOnMutableProperty.kt [hunk 2]
```diff
@@ -89,17 +89,16 @@ class NullCheckOnMutableProperty(config: Config) : Rule(
...
+                }?.getResolvedCall(bindingContext)
+                    ?.resultingDescriptor
+                    ?.fqNameOrNull()
+                    ?.takeIf(mutableProperties::contains)
+                    ?.let { candidateFqName ->
+                        // A candidate mutable property is present, so attach the current
+                        // if-expression to it in the property candidates map.
+                        candidateProperties.getOrPut(candidateFqName) { ArrayDeque() }
+                            .apply { add(expression) }
+                    }
```
Comment (cortinico) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/NullCheckOnMutableProperty.kt:101]: Same as other comment. Looks like this change is unnecessary (let's move it to another PR?)

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryLet.kt [hunk 3]
```diff
@@ -48,12 +49,14 @@ class UnnecessaryLet(config: Config) : Rule(
     "The `let` usage is unnecessary."
 ) {
 
+    class Dialog(val window: Int?)
+    val dialog: Dialog? = null
```
Comment (cortinico) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryLet.kt:53]: This should go

**Review by cortinico — CHANGES_REQUESTED** [review]

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/NullCheckOnMutableProperty.kt (hunk 2 above)
Comment (atulgpt) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/NullCheckOnMutableProperty.kt:101]: With this rule, this code is started flagging. As we have written the code as `<expression>?.let { it.<someOtherNullableChain> }` that could be simplified as `<expression>?.<someOtherNullableChain>`

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeVal.kt (hunk 1 above)
Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeVal.kt:149]: Same comment

> With this rule, this code is started flagging. As we have written the code as `<expression>?.let { it.<someOtherNullableChain> }` that could be simplified as `<expression>?.<someOtherNullableChain>`

**Review by cortinico — APPROVED** [review]
LGTM 👍 sorry for the late reply.

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**